src/gui/window/scene/note_list_system/mod.rs

```diff
@@ -120,64 +121,69 @@ impl NoteRenderer {
                 let buffer_writer = UnsafeSyncCell::new(buffer.write().unwrap());
 
                 // A system to write multiple note columns into 1 large allocated array in parallel
-                let written_notes = self.thrad_pool.install(|| {
+                let (written_notes, poly) = self.thrad_pool.install(|| {
                     // For each note column, write it into the buffer
-                    let written_notes_per_key =
-                        columns_view_info.par_iter_mut().rev().map(|column| {
-                            if column.remaining == 0 {
-                                return 0;
-                            }
-
-                            let offset =
-                                (column.offset as i64 - notes_pushed as i64).max(0) as usize;
-
-                            if offset >= buffer_length {
-                                return 0;
-                            }
-
-                            let remaining_buffer_space = buffer_length - offset;
-                            let iter_length = column.remaining;
-
-                            let allowed_to_write = if iter_length > remaining_buffer_space {
-                                remaining_buffer_space
-                            } else {
-                                iter_length
-                            };
-
-                            unsafe {
-                                let buffer = buffer_writer.get_mut();
-
-                                for i in 0..allowed_to_write {
-                                    let next_note = column.iter.next();
-                                    if let Some(note) = next_note {
-                                        buffer[i + offset] = NoteVertex::new(
-                                            note.start,
-                                            note.len,
-                                            column.key,
-                                            note.color.as_u32(),
-                                            column.border_width as u32,
-                                        );
-
-                                        if note.start <= 0.0
-                                            && column.color.is_none()
-                                            && note.start + note.len > 0.0
-                                        {
+                    let out_data = columns_view_info.par_iter_mut().rev().map(|column| {
```

Comment:
this is getting complex enough where I think a struct is better. Define a simple struct somewhere above, use it here to pass the data back.